adafruit_ble/advertising/__init__.py

```diff
@@ -271,14 +285,41 @@ def rssi(self):
            from `BLERadio.start_scan()`. (read-only)"""
         return self._rssi
 
+    @classmethod
+    def get_prefix_bytes(cls):
+        """Return a merged version of match_prefixes as a single bytes object,
+        with length headers.
+        """
+        # Do merge once and memoize it.
+        if cls._prefix_bytes is None:
+            cls._prefix_bytes = (
+                b""
+                if cls.match_prefixes is None
+                else b"".join(
+                    len(prefix).to_bytes(1, "little") + prefix
+                    for prefix in cls.match_prefixes
+                )
+            )
```

Comment:
Check for `prefixes` here and use it if available. That way you can decouple this change with the other lib changes.

Reply:
I was thinking of that, but I am changing all of our libraries that used `prefix`, and decided there would be very few users who were using `prefix` already, and was willing to not have backwards compatibility. The ones I knew of were already interested in this change. Similar thinking to the HID generalization you did.

Reply:
I'd still recommend to have it briefly. Otherwise you need to make sure and hit the same bundle.

Reply:
Added. Alternatively we could check for `.prefix` and throw an exception asking people to upgrade.